docs/topics/gradle/gradle-compiler-options.md

```diff
@@ -43,40 +87,42 @@ tasks.named('compileKotlin', org.jetbrains.kotlin.gradle.tasks.KotlinCompilation
 </tab>
 </tabs>
 
-### Target the JVM
+### Target level
 
-JVM compilation tasks are called `compileKotlin` for production code and `compileTestKotlin`
-for test code. The tasks for custom source sets are named according to their `compile<Name>Kotlin` patterns.
-
-The names of the tasks in Android Projects contain [build variant](https://developer.android.com/studio/build/build-variants.html)
-names and follow the `compile<BuildVariant>Kotlin` pattern, for example, `compileDebugKotlin` or `compileReleaseUnitTestKotlin`.
-
-For both the JVM and Android projects, it's possible to define options using the project Kotlin extension DSL:
-
-<tabs group="build-script">
-<tab title="Kotlin" group-key="kotlin">
+You can configure compiler options for the JVM/Android and Kotlin Multiplatform targets
+in the `compilerOptions {}` block inside the `target {}` block:
 
 ```kotlin
 kotlin {
-    compilerOptions {
-        apiVersion.set(org.jetbrains.kotlin.gradle.dsl.KotlinVersion.%gradleApiVersion%)
+    target { 
+        compilerOptions {
+            optIn.add("kotlin.RequiresOptIn")
+        }
     }
 }
 ```
 
-</tab>
-<tab title="Groovy" group-key="groovy">
+### Compilation unit level
 
-```groovy
-kotlin {
+You can configure compiler options for a specific compilation unit or task in a `compilerOptions {}` 
```

Comment:
Maybe we could also mention here how to access and configure via `KotlinCompilation`:
```
kotlin {
    jvm {
        val main by compilations.getting {
            compileTaskProvider.configure {
                compilerOptions {

                }
            }
        }
    }
}
```